onebusaway-android/src/main/java/org/onebusaway/android/ui/TripPlanFragment.java

```diff
@@ -127,6 +136,45 @@ interface Listener {
 
     private FirebaseAnalytics mFirebaseAnalytics;
 
+    //sets the address in the text view from the user selected contact
+    private final String ID = "id";
+    private ActivityResultLauncher<TextView> mSelectAddressFromContactLauncher = registerForActivityResult(
+            new ActivityResultContract<TextView, Intent>() {
+                int tvId;
+
+                @NonNull
+                @Override
+                public Intent createIntent(@NonNull Context context, TextView textView) {
+                    Intent intent = new Intent(Intent.ACTION_PICK);
+                    intent.setType(ContactsContract.CommonDataKinds.StructuredPostal.CONTENT_TYPE);
+                    tvId = textView.getId();
+                    return intent;
+                }
+
+                @Override
+                public Intent parseResult(int i, @Nullable Intent intent) {
+                    if (intent != null) {
+                        return intent.putExtra(ID, tvId);
+                    }
+                    return null;
+                }
+            },
+            addressIntent -> {
+                if (addressIntent == null) {
+                    return;
+                }
+                Uri addressUri = addressIntent.getData();
+                String[] projection = new String[]{ContactsContract.CommonDataKinds.StructuredPostal.FORMATTED_ADDRESS};
+                Cursor cursor = getContext().getContentResolver().query(addressUri, projection, null, null, null);
```

Comment:
The Cursor is not being closed, which can lead to memory leaks. Always close the Cursor in a finally block or use a try-with-resources statement.
```Java
try (Cursor cursor = getContext().getContentResolver().query(addressUri, projection, null, null, null)) {
    if (cursor != null && cursor.moveToFirst()) {
        int addressIndex = cursor.getColumnIndex(ContactsContract.CommonDataKinds.StructuredPostal.FORMATTED_ADDRESS);
        String address = cursor.getString(addressIndex).replace("\n", ", ");
        TextView tv = getActivity().findViewById(addressIntent.getIntExtra(ID, -1));
        tv.post(() -> tv.setText(address));
    }
}

```